### PR TITLE
feat: add spec and parser of mdadm_d

### PR DESCRIPTION
**Pull request by JoySnow**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

Add spec to collect command "/usr/sbin/mdadm -D /dev/md*"

### CODE REVIEW

**Comment by JoySnow** [comment]
@xiangce , could you help review the changes?

**Comment by JoySnow** [comment]
@xiangce , the changes are ready for review now.

**Comment by JoySnow** [comment]
Hi @xiangce , ready for review with the updates.

**Review by xiangce — APPROVED** [review]

File: insights/parsers/mdadm.py [hunk 1]
```diff
@@ -65,3 +76,143 @@ def parse_content(self, content):
...
+
+        has_device_table = bool(table_start_index > device_start_index)
+
+        kv_pairs_end_index = table_start_index if has_device_table else index
+        # Parse the key value pairs part in content
+        if kv_pairs_end_index - device_start_index > 1:
+            self.update(split_kv_pairs(content[device_start_index + 1:kv_pairs_end_index], split_on=':'))
+
+        # Parse the devices info table part in content
+        self.devices_table = parse_fixed_table(content[table_start_index:index]) if has_device_table else None
```
[the remark this answers is not in the document]
Comment (JoySnow) [reply insights/parsers/mdadm.py]: Updated the default value to `[]`

**Review by xiangce — APPROVED** [review]

**Pull request merged**